packages/ssm-cache/index.ts

```diff
@@ -14,33 +14,50 @@ export type SsmCache = {
 
 export const ssmCache: SsmCache = { values: {} };
 
-export const addToCache = (regex, { Name = null, Value = null }: SSM.Parameter) => {
+export const getSsmParameter = (name: string): string => {
+  if (!Object.prototype.hasOwnProperty.call(ssmCache.values, name)) {
+    throw new Error(`SSM parameter ${name} not found in cache`);
+  }
+
+  return ssmCache.values[name] || '';
+};
+
+export const addToCache = (regex: RegExp | undefined, { Name, Value }: SSM.Parameter) => {
   if (!Name) return;
   if (regex && !regex.test(Name)) return;
 
   ssmCache.values[Name] = Value;
 };
 
-const hasCacheExpired = () => !!(ssmCache.expiryDate && new Date() > ssmCache.expiryDate);
+export const hasCacheExpired = () => !!(ssmCache.expiryDate && new Date() > ssmCache.expiryDate);
+
+export const isConfigNotEmpty = () => !!Object.keys(ssmCache.values).length;
+
+export const getSsmClient = () => {
```

Comment:
We want to initialize ssm when we need it so we can mock it with jest.

Reply:
I was wondering if we should just have this module expose a factory rather than expose a singleton, and let the caller manage the instance, rather than enforcing a singleton here?

I appreciate in the lambda use case that might mean we just have a stub module with a field & a getter in it, but if we are going to use it in the HRM service too, leaving it up to the caller might work better?

I'm just a bit wary of module level fields in dependencies in node, you saw from my serverless experiments how there are scenarios where you could potentially end up with multiple 'singletons' if you aren't careful...

Reply:
Duplicate "singletons" wouldn't really matter all that much in this use case. The `getSsmClient()` function really only exists for call-time instantiation so that we can properly mock `aws-sdk.SSM` in tests. It seems like it may be overkill to manage a singleton in the caller for this case. We could technically call `new SSM()` on each call to getSsmClient without incurring much overhead or impacting functionality.

Reply:
Also, I think I missed the serverless experiments you are referencing. Could you point me in the right direction to review those?